third_party/move/tools/move-asm/src/disassembler.rs

```diff
@@ -0,0 +1,705 @@
+                    .write_str(&abilities.display_postfix().to_string())?;
+            },
+            Struct(idx) | StructInstantiation(idx, _) => {
+                let view = StructHandleView::new(module, module.struct_handle_at(*idx));
+                write!(
+                    self.out,
+                    "{}{}",
+                    self.module_id_prefix(&view.module_id())?,
+                    view.name()
+                )?;
+                if let StructInstantiation(_, inst) = ty {
+                    self.list(inst, |dis, _, e| dis.type_(module, e), "<", ",", ">")?
+                }
+            },
+            Reference(elem_ty) => {
+                self.out.write_str("&")?;
+                self.type_(module, elem_ty)?
+            },
+            MutableReference(elem_ty) => {
+                self.out.write_str("&mut ")?;
+                self.type_(module, elem_ty)?
+            },
+            TypeParameter(idx) => self.out.write_str(&type_param_name(*idx as usize))?,
+        }
+        Ok(())
+    }
+
+    // --------------------------------------------------------------------------------------
+    // Functions
+
+    fn fun(&mut self, fdef: FunctionDefinitionView<CompiledModule>) -> anyhow::Result<()> {
+        // Function header
+        if fdef.is_entry() {
+            self.out.write_str("entry ")?
+        }
+        match fdef.visibility() {
+            Visibility::Private => {},
+            Visibility::Public => self.out.write_str("public ")?,
+            Visibility::Friend => self.out.write_str("friend ")?,
+        }
+        write!(self.out, "fun {}", fdef.name())?;
+        if !fdef.type_parameters().is_empty() {
+            self.list(
+                fdef.type_parameters().as_slice(),
+                |dis, idx, abilities| dis.type_param_decl(false, idx, *abilities),
+                "<",
+                ",",
+                ">",
+            )?
+        }
+        let arg_tokens = fdef.arg_tokens().collect::<Vec<_>>();
+        self.list(
+            &arg_tokens,
+            |dis, pos, ty| {
+                write!(dis.out, "{}: ", local_name(pos as LocalIndex))?;
+                dis.type_(fdef.module(), ty.signature_token())
+            },
+            "(",
+            ",",
+            ")",
+        )?;
+        if fdef.return_count() > 0 {
+            self.out.write_str(": ")?;
+            if fdef.return_count() > 1 {
+                self.list(
+                    &fdef.return_tokens().collect::<Vec<_>>(),
+                    |dis, _, ty| dis.type_(fdef.module(), ty.signature_token()),
+                    "(",
+                    ",",
+                    ")",
+                )?
+            } else {
+                self.type_(
+                    fdef.module(),
+                    fdef.return_tokens().next().unwrap().signature_token(),
+                )?
+            }
+        }
+        writeln!(self.out)?;
+
+        if let Some(unit) = fdef.code() {
```

Comment:
Do we plan to support `native` keyword, or using empty bodied functions is the. way to write them?